pallets/parachain-staking/src/mock.rs

```diff
@@ -253,6 +257,41 @@ pub(crate) fn events() -> Vec<pallet::Event<Test>> {
 		.collect::<Vec<_>>()
 }
 
+/// Prints the diff iff assert_eq fails
+#[macro_export]
+macro_rules! asserts_eq {
+	($left:expr, $right:expr) => {
+		match (&$left, &$right) {
+			(left_val, right_val) => {
+				similar_asserts::assert_eq!(*left_val, *right_val);
+			}
+		}
+	};
+}
```

Comment:
`assert_eq` will just say the two sides are not equal and print out each side.

It does not highlight the diff between each side. This macro highlights the diff so it is easier to debug expected events compared to actual events in the unit tests. (I'll show it in this comment soon)

Reply:
This macro was actually replaced by other test utility macros: `assert_last_event`, `assert_event_emitted`, `assert_eq_events`. These have to be macros because we need the panic information to be emitted at the call site and when these are functions, all errors point to `mock.rs` instead of `tests.rs` because macros are expanded upon invocation at the place where they're invoked.